Homework/homework1/task4.py

```diff
@@ -0,0 +1,35 @@
+"""
+Classic task, a kind of walnut for you.
+
+Given four lists A, B, C, D of integer values,
+    compute how many tuples (i, j, k, l) there are such that A[i] + B[j] + C[k] + D[l] is zero.
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+from typing import List, Mapping
+
+
+def create_dict(x: int, x_dict: Mapping[int, int]) -> Mapping[int, int]:
+    """Add repeated elements into the dictionary."""
+    if x in x_dict:
+        x_dict[x] += 1
+    else:
+        x_dict[x] = 1
+    return x_dict
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:
```

Comment:
What is alg and memory complexity of this solution?

Reply:
Alg complexity is O(n^2)
About memory Im not sure how to calculate it properly.. Input is 4N, than 2N^2 for two lists and 4N^2 for two dicts for the worst case. So I assume its also O(n^2) because there is a quadratic relation between input and other memory.

Reply:
Input and output are ignored when calculating memory.  So only extra memory by alroritm itself coounts.